### PR TITLE
Persisted queries: Comment update suggested in #3566 review

### PR DESCRIPTION
**Pull request by glasser**
This feedback came in right before the PR was merged.

### CODE REVIEW

**Review by abernix — APPROVED** [review]

**Comment by abernix** [comment]
Hmm, tests failed with:

```
---- configuration::tests::validate_project_config_files stdout ----
thread 'configuration::tests::validate_project_config_files' panicked at './src/configuration/testdata/metrics/persisted_queries.router.yaml configuration error: 
could not deserialize configuration: apqs must be disabled to enable safelisting: either set preview_persisted_queries.safelist.enabled: false or apq.enabled: false in your router yaml configuration', apollo-router/src/configuration/tests.rs:451:21
```


... which is a test that we have that validates that configuration in our documentation actually works.  Does that sound like a problem based on what you know about where our documentation is? I was just strolling through here.

**Comment by abernix** [comment]
I merged in `dev`, just in case that's the fix.

**Comment by glasser** [comment]
That does seem like a test that could be relevant to this change or #3566 (I did change some of these error messages late in the game), but I would have assumed it would have been caught there! Will take a look.

**Comment by glasser** [comment]
This looks like a race between https://github.com/apollographql/router/pull/3595 and #3566 that @BrynCooke fixed during merge commit 4b52959a. I think your merge from dev probably fixed it.

(You might want to consider enabling GitHub's merge queue feature, which helps ensure that overlapping PRs merged in parallel don't break each other, by running tests against the merge result before allowing the merge to land. We use it happily in our internal studio-ui project.)

**Comment by abernix** [comment]
Looks like it is indeed working onw.  I looked at enabling Merge Queues before, but there were a couple gotchas and one immediate blocker with branch protection rule name rules (wildcards aren't allowed or something?) that made me move on.  Doesn't happen all that often.  Thanks!

**Review by o0Ignition0o — APPROVED** [review]

File: apollo-router/src/services/layers/persisted_queries/manifest_poller.rs [hunk 1]
```diff
@@ -50,8 +50,10 @@ pub(crate) enum FreeformGraphQLBehavior {
     },
 }
 
-/// Describes what the router should do for a given request: allow it, deny it
-/// with an error, or allow it but log the operation as unknown.
+/// How we should treat requests containing freeform GraphQL rather than a PQ ID.
+/// (This also includes the results of APQ lookups, for the purpose of differenting
+/// between Allow and AllowAndLog; APQ cannot be enabled in any mode that allows
+/// freeform requests to be denied.)
 pub(crate) enum FreeformGraphQLAction {
```
Comment (garypen) [inline apollo-router/src/services/layers/persisted_queries/manifest_poller.rs:57]: This isn't what I was suggesting. The thrust of my suggested change was that the description of each enum variant was associated directly with it. If you run `cargo doc --open` you can see the impact of the difference between documenting as is (or with this change) vs documenting as this:
```
/// How should we treat Freeform GraphQL
pub(crate) enum FreeformGraphQLAction {
    /// Allow it
    Allow,
    /// Deny it, with an error
    Deny,
    /// Allow it, but log the operation as unknown
    AllowAndLog,
    /// Deny it, with an error and log the operation as unknown <- Presumably...
    DenyAndLog,
}
```
The documentation of each variant is explicitly with the variant and it's impossible for the documentation of the enum and the variants to be misconstrued or get out of sync over time.

**Review by garypen — CHANGES_REQUESTED** [review]

**Comment by glasser** [comment]
Sorry, doesn't look like I'm getting to improving this until after I'm back from next week's vacation based on the amount of time I have left this week.

**Comment by Geal** [comment]
should we close this or finish it?

**Comment by abernix** [comment]
I'll close this since we haven't figured out next steps, but it's more than possible to re-open it.

**Pull request closed without merging**